dvc/remote/gdrive.py

```diff
@@ -325,10 +323,37 @@ def _create_remote_dir(self, parent_id, title):
 
     @gdrive_retry
     def _delete_remote_file(self, remote_id):
+        from pydrive2.files import ApiRequestError
+
         param = {"id": remote_id}
         # it does not create a file on the remote
         item = self.drive.CreateFile(param)
-        item.Trash() if self._gdrive_trash_only else item.Delete()
+
+        try:
+            item.Trash() if self._gdrive_trash_only else item.Delete()
+        except ApiRequestError as exc:
+            if self.list_params["corpora"] != "drive" or not exc.args:
+                raise
+            http_error = exc.args[0]
+            if (
+                http_error.resp.status == 403
+                and "Insufficient permissions for this file".encode("utf8")
```

Comment:
I thought that we agreed on changing this on the PyDrive2 end? To fix the issue with retries, and avoid using string parsing.

Reply:
`403` code seems is not an unique identifier to detect insufficient permissions error; it is required to check additionally http_error.content which has JSON string in bytes format with `message` key equal to "Insufficient permissions for this file".
Since we discussed that it doesn't make sense to introduce separate exception class in PyDrive2 along with ApiRequestError, it seems a simple alternative way just to handle this on DVC side by alanyzing 2 unique params (used string search might be replaced into more complex converting from bytes array to a map and checking key value).
Please let me know if I'm still missing here a better way to handle this, thanks!

Reply:
as we discussed, Google Drive returns `reason` for every error it sends in the JSON structure - https://developers.google.com/drive/api/v3/handle-errors . We should be parsing this JSON, extracting status code, reason, all other relevant information and include it into ApiRequestError.

If the JSON you mentioned is exactly the one described in the doc and we already have it, we can parse it on our end.